source/mus_species_module.f90

```diff
@@ -275,12 +291,90 @@ subroutine mus_load_species( me, conf, parent, minLevel, nFields, physics, &
+
+    if ( sub_handle /= 0 ) then
+      call aot_get_val( L = conf,                                              &
+        &               thandle = sub_handle,                                  &
+        &               key = 'Dxx',                                           &
+        &               val = me%diff_tensor%Dxx,                              &
+        &               ErrCode = iError,                                      &
+        &               default = 0.0_rk )
+
+      call aot_get_val( L = conf,                                              &
+        &               thandle = sub_handle,                                  &
+        &               key = 'Dyy',                                           &
+        &               val = me%diff_tensor%Dyy,                              &
+        &               ErrCode = iError, &
+        &               default = 0.0_rk )
+      
+      call aot_get_val( L = conf,                                              &
+        &               thandle = sub_handle,                                  &
+        &               key = 'Dzz',                                           &
+        &               val = me%diff_tensor%Dzz,                              &
+        &               ErrCode = iError,                                      &
+        &               default = 0.0_rk )
+
+      call aot_get_val( L = conf,                                              &
+        &               thandle = sub_handle,                                  &
+        &               key = 'Dxy',                                           &
+        &               val = me%diff_tensor%Dxy,                              &
+        &               ErrCode = iError,                                      &
+        &               default = 0.0_rk )
+
+      call aot_get_val( L = conf,                                              &
+        &               thandle = sub_handle,                                  &
+        &               key = 'Dxz',                                           &
+        &               val = me%diff_tensor%Dxz,                              &
+        &               ErrCode = iError,                                      &
+        &               default = 0.0_rk )
+
+      call aot_get_val( L = conf,                                              &
+        &               thandle = sub_handle,                                  &
+        &               key = 'Dyz',                                           &
+        &               val = me%diff_tensor%Dyz,                              &
+        &               ErrCode = iError,                                      &
+        &               default = 0.0_rk )
+
+      if (me%diff_tensor%Dxx == 0.0_rk .and. me%diff_tensor%Dyy == 0.0_rk .and.     &
+        &    me%diff_tensor%Dzz == 0.0_rk .and. me%diff_tensor%Dxy == 0.0_rk .and.  &
+        &    me%diff_tensor%Dxz == 0.0_rk .and. me%diff_tensor%Dyz == 0.0_rk ) then
```

Comment:
If `diff_tensor` where an array this could simply be: `if (all(diff_tensor == 0.0_rk)) then`, or better yet exchange the branches and put the erroneous condition second with `if (any(diff_tensor != 0.0_rk)) then` (the compiler may complain about equality comparisons for real numbers).